internal/executor/executor.go

```diff
@@ -0,0 +1,168 @@
+package executor
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Executor interface lets us implement dummy executors for tests
+type Executor interface {
+	ExecWithContext(context.Context) (*Result, error)
+	SetArgs([]string)
+	SetEnv([]string)
+	SetQuiet()
+	SetVerbose()
+	GetCmd() string
+	DisableBinCheck()
+	SetStdin(io.Reader)
+	CmdPath() string
+	CheckExecutable() error
+	// for tests
+	SetStdout([]byte)
+	SetStderr([]byte)
+	SetExitCode(int)
```

Comment:
Do these have to be exported?

Reply:
I'd have to review the code in more detail, but my guess would be these are exported for use in testing.    They are exported here because they were exported in the ironlib code that this was directly copied from.

Reply:
when we're using a mock library, these would not be required